### PR TITLE
Added SentryId struct

### PR DESCRIPTION
**Pull request by MercinQ**
I added a SentryId ` struct` , I hope it is what you wanted. Im just not sure where to put  the `.cs` file.

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Thanks @MercinQ for your contribution.
I second @untitaker's comments. 
Could you please change the return types of the `Capture` methods in `ISentryClient` (and whatever the compiler leads you to after that)?

File: src/Sentry/Extensibility/DisabledHub.cs [hunk 1]
```diff
@@ -21,10 +21,10 @@ public class DisabledHub : IHub, IDisposable
 
         public void BindClient(ISentryClient client) { }
 
-        public Guid CaptureEvent(SentryEvent evt, Scope scope = null) => Guid.Empty;
+        public SentryId CaptureEvent(SentryEvent evt, Scope scope = null) => new SentryId(Guid.Empty);
```
Comment (bruno-garcia) [inline src/Sentry/Extensibility/DisabledHub.cs]: With the `implicit` operator in place, there shouldn't be a need to call the constructor here.
Can you simply return: `default` instead? Here and all other occurrences please.

Comment (bruno-garcia) [reply src/Sentry/Extensibility/DisabledHub.cs]: Even better would be to create a static readonly property called `Empty` inside `SentryId`
This way you can simply use: `SentryId.Empty`

**Comment by bruno-garcia** [comment]
@MercinQ looking great! Thanks a lot for your help. I left a couple other comments and resolves all reviews that were fixed (you can click _Close review_ when you resolved the issues if you want)
Cheers!

**Comment by MercinQ** [comment]
Thank you, really much for help and guidance. This was my first participation in an open source project and I hope I can help with other issues in the future.

**Comment by bruno-garcia** [comment]
@MercinQ glad to have your help and happy to help out.
The PR is nearly done and I noticed you closed it. I hope it was just an accident and you intend to stick around so I'll reopen it for you.
Thanks again

**Comment by MercinQ** [comment]
@bruno-garcia I wasn't sure about closing this PR cause I couldn't find those comments that you previously mentioned. Can you please tell me where I can find them exactly?

**Comment by bruno-garcia** [comment]
@MercinQ I see the PR was closed again. When you `Close` a PR, I'm not able to merge it in anymore, so it's like you've given up on the work.

Since the work is almost done, I'll reopen it again.
The last few comments can be seen, as you said, as `pending` comments on this tab, or on the [`Files changed`](https://github.com/getsentry/sentry-dotnet/pull/140/files) tab.
Once again thanks for your help :)

**Comment by MercinQ** [comment]
I had to study hard lately, apologies for the delay. Hmm I wrote an answer to unresolved comments, cause I think I satisfied suggestions mentioned in them

File: src/Sentry/SentryId.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Sentry
+{
+   public struct SentryId
```
[the remark this answers is not in the document]
Comment (MercinQ) [reply src/Sentry/SentryId.cs]: Srry for the delay I had some things to take care of. I'm not sure but I feel like I solved this comment with latest commit. Can you pinpoint exactly what is still wrong here?

File: src/Sentry/SentryId.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {
+       public Guid eventID;
+
+        public override string ToString()
```
[the remark this answers is not in the document]
Comment (MercinQ) [reply src/Sentry/SentryId.cs]: Srry for the delay I had some things to take care of. I'm not sure but I feel like I solved this comment with latest commit. Can you pinpoint exactly what is still wrong here?

File: src/Sentry/SentryId.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        {
+          return eventID.ToString("n");
+        }
+
+        public static implicit operator Guid(SentryId d)
```
[the remark this answers is not in the document]
Comment (MercinQ) [reply src/Sentry/SentryId.cs]: Srry for the delay I had some things to take care of. I'm not sure but I feel like I solved this comment with latest commit. Can you pinpoint exactly what is still wrong here?

**Comment by bruno-garcia** [comment]
@MercinQ No worries. Sorry for my delay, but I had a few days off.

Please note more code was changed in master through other pull requests.
So you likely will need to pull master into your branch and make sure it builds (fix any issues) and push again.

**Comment by bruno-garcia** [comment]
@MercinQ somehow it was showing me outdated code and comments. Now comments I made are gone. It's weird.

I'll merge this in and fix anything missing. I'll tag you for a review of anything.

Thanks!!

**Pull request merged**